File: src/test/java/io/gaegul/ch01/DecoyDuckTest.java
package io.gaegul.ch01;

import static org.assertj.core.api.Assertions.assertThat;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.DisplayNameGeneration;
import org.junit.jupiter.api.DisplayNameGenerator;
import org.junit.jupiter.api.Nested;
import org.junit.jupiter.api.Test;

import io.gaegul.util.ConsoleIOTest;

@DisplayNameGeneration(DisplayNameGenerator.ReplaceUnderscores.class)
@DisplayName("오리 시뮬레이션 게임 - 가짜 오리는")
class DecoyDuckTest extends ConsoleIOTest {

	@Test
	void 헤엄을_칠_수_있다() {
		Duck duck = new DecoyDuck();
		boolean result = duck.swim();
		assertThat(result).isTrue();
	}

	@Nested
	class display_메서드를_실행하면 {

		@Test
		void 가짜오리를_출력한다() {
			Duck duck = new DecoyDuck();
			duck.display();
			assertThat(output()).isEqualTo("가짜 오리");
		}

	}
}

File: src/test/kotlin/io/gaegul/ch01/DecoyDuckTest.kt
package io.gaegul.ch01

import io.gaegul.utils.ConsoleIOTest
import org.assertj.core.api.Assertions.assertThat
import org.assertj.core.api.Assertions.assertThatThrownBy
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test

@DisplayName("오리 시뮬레이션 게임 - 가짜오리는")
class DecoyDuckTest: ConsoleIOTest() {

    private val duck: DecoyDuck
        get() {
            return DecoyDuck()
        }

    @Test
    fun `소리를 낼 수 없다`() {
        assertThatThrownBy { duck.performQuack() }
            .isInstanceOf(UnsupportedOperationException::class.java)
    }

    @Test
    fun `혜엄을 칠 수 있다`() {
        val result = duck.swim()
        assertThat(result).isTrue()
    }

    @Test
    fun `날 수 없다`() {
        assertThatThrownBy { duck.performFly() }
            .isInstanceOf(UnsupportedOperationException::class.java)
    }

    @Test
    fun `display()를 실행하면 가짜오리를 출력한다`() {
        duck.display()
        assertThat(output()).isEqualTo("가짜 오리")
    }
}